### DISCUSSION TITLE
Spec: Automated QA Strategy

### DISCUSSION THREAD

**Discussion by dan-menlo** [janhq/jan#3573] [Architecture Specs]
## Objective

- Be able to move to weekly release of Jan and Cotex w/o increase in headcount, with 0 bug reports

## Scope

- Installation and Uninstallation testing
- Upgrade testing
- Functionality testing

**Comment by Van-QA** [comment]
Hi @dan-homebrew,
Notion spec for the discussion: https://www.notion.so/jan-ai/Automated-QA-Strategy-a53e1e72445343398d1d519a5ffe7084?pvs=4

Linking related tickets, TBU to cover more: https://github.com/orgs/janhq/projects/5/views/53?sliceBy%5Bvalue%5D=Automated+Testing
### Unit test:
- [ ] https://github.com/janhq/jan/issues/3495
- [ ] https://github.com/janhq/jan/issues/3507
- [ ] https://github.com/janhq/cortex/issues/1069

### API test:
- [ ] https://github.com/janhq/jan/issues/3547
- [ ] https://github.com/janhq/cortex/issues/1067 => To be discussed, to cover more hardware

### E2E config:
- [ ] https://github.com/janhq/jan/issues/2047
- [ ] https://github.com/janhq/jan/issues/3510

- [ ] epic: Monkey testing for Jan app
- [ ] feat: run e2e test in multiple screensize

### E2E test
- [ ] https://github.com/janhq/jan/issues/3556
- [ ] feat: Sidebar configuration for threads settings reflect in API request


To be updated for more tickets
